clients/src/main/java/org/apache/kafka/common/metrics/JmxReporter.java

```diff
@@ -318,4 +321,15 @@ public AttributeList setAttributes(AttributeList list) {
                                       + ".(whitelist/blacklist) is not a valid regular expression");
         }
     }
+
+    @Override
+    public void contextChange(MetricsContext metricsContext) {
+        Objects.requireNonNull(metricsContext.metadata().get(MetricsContext.NAMESPACE));
+        synchronized (LOCK) {
+            if (!this.mbeans.isEmpty()) {
```

Comment:
nit, we don't use `this.` anywhere else in this code when referring to mbean. It would be good to stay consistent.